tgui/packages/tgui/interfaces/common/PressureBar.js

```diff
@@ -0,0 +1,52 @@
+import { Box, ProgressBar, AnimatedNumber } from '../../components';
+
+export const PressureBar = props => {
+  const {
+    pressure,
+    maxPressure,
+  } = props;
+
+  const bgColorSafe = "black";
+  const bgColorDanger = "red";
+  const barColorSafe = "green";
+  const barColorDanger = "yellow";
+
+  const barColor = () => {
+    if ((pressure / maxPressure) > 1) {
+      return barColorDanger;
+    } else {
+      return barColorSafe;
+    }
+  };
+
+  const bgColor = () => {
+    if ((pressure / maxPressure) > 1) {
+      return bgColorDanger;
+    } else {
+      return bgColorSafe;
+    }
+  };
+
+  const pct = () => {
+    if ((pressure / maxPressure) > 1) {
+      return pressure / (maxPressure * 10);
+    } else {
+      return pressure / maxPressure;
+    }
+  };
+
+  return (
+    <Box
+      danger p={0.5}
+      backgroundColor={bgColor()}>
+      <ProgressBar
+        color={barColor()}
+        value={pct()}>
```

Comment:
Use of these functions without arguments in here feels a little weird. I understand what it's doing but seems like we could either a) make these util functions that take arguments, or b) just have `barColor`/`pct` be the result rather than the function itself.

Reply:
fixed